### DISCUSSION TITLE
Option 125 - send from client to server

### DISCUSSION THREAD

**Discussion by suravakonda** [rsmarples/dhcpcd#169] [Q&A] [answered]
Dear all,

Recently I've switched from dhclient to dhcpcd on my client platform. Everything works as it should, except the Option 125 information that I need to send from client to server (ISC DHCP server).

I am trying to send it as below from dhcpcd.conf

vendor 125, ab:cd:ef:....hex string....

This is appearing in wireshark as Option 43 (encapsulated vendor option)

I now realise I need something like this (as given in the dhcpcd.conf man page)

define 125 encap vsio
embed uint32 enterprise_number       (I have an enterprise number for my device)
encap 1 data string

When I send data like this, I am not able to see option 125 on wireshark trace.

In one of the older threads (from 2018) Mr Marples suggested that the option send from client might not work properly. Has anyone attempted this and knows it works?  (Related link here: https://roy.marples.name/archives/dhcpcd-discuss/0002220.html )

Thanks for reading this far...I will keep trying but any pointers will be helpful.

*** EDIT ***
After having re-read the conf file documentation at https://www.daemon-systems.org/man/dhcpcd.conf.5.html

I have now tried with following lines at end of dhcpcd.conf, to test a sample option 125.

define 125 encap vsio
embed uint32 1234
encap 1 ipaddress 01234567

But still I am not seeing dhcpcd send out option 125 packet to the server (seen via Wireshark logs).

Please can I get some inputs on how to construct option 125 to be sent from dhcpcd client -> dhcp server?

**Comment by rsmarples** [comment, marked as answer]
Currently dhcpcd can only decode the DHCP option 125, it cannot send it. No-one has asked for the ability to send it before, just request it and decode it. Although RFC 3925 allows the sending of it, it's heavily implied that the client sends option 124 and receives option 125.

**Reply by suravakonda** [reply]
Thanks for your reply Roy. If it's worth implementing, I can give a try. When you get a few minutes, can you please point me to an existing "option 124 client send". I will have a look and give it a go to implement option 125. Thanks.

**Reply by rsmarples** [reply]
vivco options are parsed here:
https://github.com/NetworkConfiguration/dhcpcd/blob/master/src/if-options.c#L1959
vivco dhcp is assembled here:
https://github.com/NetworkConfiguration/dhcpcd/blob/master/src/dhcp.c#L1061
vivco dhcp6 is assembled here:
https://github.com/NetworkConfiguration/dhcpcd/blob/master/src/dhcp6.c#L269
https://github.com/NetworkConfiguration/dhcpcd/blob/master/src/dhcp6.c#L302

**Comment by ido** [comment]
Resolving this discussion for now. If you decide to submit a PR, I'll take a look there.
